core/src/main/java/io/kestra/core/utils/GraphUtils.java

```diff
@@ -0,0 +1,455 @@
+package io.kestra.core.utils;
+
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.executions.Execution;
+import io.kestra.core.models.executions.TaskRun;
+import io.kestra.core.models.flows.Flow;
+import io.kestra.core.models.hierarchies.*;
+import io.kestra.core.models.tasks.FlowableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.tasks.flows.Dag;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GraphUtils {
```

Comment:
I didn't check this class, did you change some methods from the previous GraphService that I may have a look of is it the exact same?

Reply:
Yes a lot, mainly the fillGraph method :'(